### ISSUE TITLE
Refactor `sct_deepseg` to store model orientation in model dict

### ISSUE THREAD

**Issue by joshuacwnewton** [spinalcordtoolbox/spinalcordtoolbox#4450]
It seems like your model was trained [using the AIL orientation](https://github.com/ivadomed/canproco/blob/da39ff9b3c3886b3ad931f5d8baf4963e45a9a2c/packaging/run_inference_single_subject.py#L91-L97)? If that's the case, then we will need to modify the `sct_deepseg` code, since it currently assumes that nnunetv2 models were trained using LPI orientation:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/8f2b0e0eb956cbd12d1cabdfbee34e1ca8d99c34/spinalcordtoolbox/deepseg/inference.py#L198-L202

Perhaps it would be best to add "orientation" to the `MODELS` dict for each of the nnunet/monai models, then specify LPI/RPI/AIL on a per-model basis, then fetch this value from the model dict when reorienting the input image?

_Originally posted by @joshuacwnewton in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4443#discussion_r1571210940_

**Comment by jcohenadad** [comment]
> Perhaps it would be best to add "orientation" to the MODELS dict for each of the nnunet/monai models, then specify LPI/RPI/AIL on a per-model basis, then fetch this value from the model dict when reorienting the input image?

Couldn't we encode the reorientation step directly from MONAI's data preparation phase, while running the inference? Eg: code like [this one](https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/blob/32ef5c9d541bb779453cca6fc8c25b768bc2c9b1/monai/run_inference_single_image_v2.py#L89) is supposed to take care of this, no? Tagging @naga-karthik @plbenveniste @Nilser3 @NathanMolinier

**Comment by NathanMolinier** [comment]
> code like [this one](https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/blob/32ef5c9d541bb779453cca6fc8c25b768bc2c9b1/monai/run_inference_single_image_v2.py#L89) is supposed to take care of this, no?

Yes it is indeed the case, however I don't know if nnUNet is performing any reorientations beforehand.

Also, I wanted to remind that [conventions](https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4419) may be different depending on the tools that were used: reorientation could be done before the training as pre-processing using SCT for example or using MONAI's transform.

**Comment by NathanMolinier** [comment]
I did not implement any model yet to SCT, but aren't we calling model specific inference scripts when calling the models ?

**Comment by joshuacwnewton** [comment]
Sorry, yes! I didn't clarify, but this is indeed specific to nnunetv2 (i.e. Not monai) models
